### ISSUE TITLE
Inference compiles the model twice for a flag the refiner ignores

### ISSUE THREAD

**Issue by MatthewLacerda2** [MatthewLacerda2/TinyRefinementModel#207]
**Lane:** cpu — the defect is a compile-count, which is measurable on CPU.

## What's wrong

`trm/infer.py:90` marks `refresh` as a **static** jit argument:

```python
@partial(nnx.jit, static_argnames=['refresh', 'top_k', 'top_p', 'depth'])
def get_logits_for_token(model, padded_tks, token_idx, refresh, top_k, top_p, temperature, depth):
```

and `trm/infer.py:125` flips it every 4 tokens:

```python
new_document = (i % HUNCH_REFRESH_EVERY == 0)
```

`refresh` becomes `new_document`, which `RefinerForTraining.__call__` **documents that it ignores** — Plan A carries no state between windows. So a flag with no effect on the computation is part of the jit cache key, and we build **two distinct executables for the same computation**.

Measured on a toy-dim refiner (CPU, counting `compile_or_get_cached` calls):

```
XLA compilations over 8 generated tokens, refresh alternating : 2
XLA compilations over 8 generated tokens, refresh pinned True : 0
```

On the card that is two resident executables and two sets of CUDA graphs. `trm/config.py` already blames exactly that class of driver-memory pressure — "28 alive graphs ... the driver cannot instantiate a CUDA command buffer" — for squeezing batch-2 from *outside* the BFC arena.

`HUNCH_REFRESH_EVERY` (`trm/infer.py:26`) is reasoner-era plumbing; the hunch is proven inert (`docs/findings/2026-06-13-cross-window-hunch-inert.md`) and the default arch is `refiner`.

## This is not #204

#204 (persistent compilation cache) is real and worth doing, but it is a different problem. A perfect cache would load **two** executables from disk instead of compiling two — the duplication is in the cache key, not in the cold start. Fix here is deleting the staticness, not caching around it.

## Note on #153

#153 flags this same plumbing under "While in there". Splitting it out because it is a one-line change that lands independently and immediately, while #153 is a multi-day rewrite. Drop that bullet from #153 when this merges.

## Acceptance

- **CPU, blocking:** a test asserting that generating N tokens triggers **one** compilation of the sampling step, not two. Goes in `tests/core/`.
- Generated text for a fixed prompt + seed is unchanged (the flag was inert, so this must be exact).
- Decide the fate of `HUNCH_REFRESH_EVERY` for the `reasoner` path rather than deleting it blind — it is the control baseline and does read `new_document`.
